src/main/java/com/mindee/extraction/ImageExtractor.java

```diff
@@ -52,14 +74,67 @@ public <FieldT extends PositionData> List<ExtractedImage> extractImages(List<Fie
 
   /**
    * Extract images from a list of fields having position data.
+   * Use this when the input file is an image or a single-page PDF.
    * @param fields List of Fields to extract.
-   * @param filename The base output filename.
+   * @param outputName The base output filename, must have an image extension.
    * @return A list of {@link ExtractedImage}.
    */
-  public <FieldT extends PositionData> List<ExtractedImage> extractImages(List<FieldT> fields, String filename) {
+  public <FieldT extends PositionData> List<ExtractedImage> extractImages(List<FieldT> fields, String outputName) {
+    if (this.getPageCount() > 1) {
+      throw new MindeeException("Input file has more than one page, use the `extractPageImages` method instead.");
+    }
+    return extractFromPage(fields, 0, outputName);
+  }
```

Comment:
Why not just use the same function for both? Separating single vs multi-page seems like an easy way to get things mixed-up during usage, i.e. if you're treating batches of file

Reply:
right you are, done.